source/mir/random/algorithm.d

```diff
@@ -34,868 +23,224 @@ Params:
+    return randomSlice(rne, var, lengths);
 }
 
 /// ditto
-auto randomSlice(G, size_t N)(G gen, size_t[N] lengths...)
-    if (isSaturatedRandomEngine!G &&
-        (is(G == class) || is(G == interface)))
-{
-    import mir.ndslice.allocation: slice;
-    import mir.ndslice.slice: slicedField;
-    return field(gen).slicedField(lengths).slice;
-}
-
-/// ditto
-auto randomSlice(G, size_t N)(G* gen, size_t[N] lengths...)
-    if (isSaturatedRandomEngine!G &&
-        !is(G == class) && !is(G == interface))
-{
-    import mir.ndslice.allocation: slice;
-    import mir.ndslice.slice: slicedField;
-    return field(gen).slicedField(lengths).slice;
-}
-
-/// ditto
-auto randomSlice(G, size_t N)(ref G gen, size_t[N] lengths...)
-    if (isSaturatedRandomEngine!G &&
-        !is(G == class) && !is(G == interface))
-{
-    import mir.ndslice.allocation: slice;
-    import mir.ndslice.slice: slicedField;
-    return field(gen).slicedField(lengths).slice;
-}
-
-/// ditto
-auto randomSlice(alias gen = rne, size_t N)(size_t[N] lengths...)
-    if (__traits(compiles, { static assert(isSaturatedRandomEngine!(typeof(gen))); }))
-{
-    import mir.ndslice.allocation: slice;
-    import mir.ndslice.slice: slicedField;
-    return field!gen.slicedField(lengths).slice;
-}
-
-/// Normal distribution
-nothrow @safe version(mir_random_test) unittest
-{
-    import mir.ndslice: slicedField, slice;
-    import mir.random.variable: NormalVariable;
-    auto sample = NormalVariable!double(0, 1).randomSlice(100);
-
-    assert(sample.shape == [100]);
-
-    import mir.ndslice.slice;
-    assert(is(typeof(sample) == Slice!(double*)));
-}
-
-/// Normal distribution for complex numbers
-nothrow @safe version(mir_random_test) unittest
-{
-    import mir.ndslice: slicedField, slice;
-    import mir.random;
-    import mir.random.variable: NormalVariable;
-
-    //Using pointer to RNG:
-    auto var = NormalVariable!double(0, 1);
-    auto sample1 = threadLocalPtr!Random.randomSlice(var, 5, 3);
-
-    //Using alias of default thread local Random Engine (`rne`):
-    auto sample2 = randomSlice!cdouble(var, 5, 3);
-
-    Random rng = Random(12345);
-    auto sample3 = rng.randomSlice!cdouble(var, 5, 3); //rng safely passed by ref
-}
-
-/// Normal distribution for complex numbers
-nothrow @safe version(mir_random_test) unittest
-{
-    import mir.ndslice: slicedField, slice;
-    import mir.random;
-    import mir.random.variable: NormalVariable;
-
-    //Using pointer to RNG:
-    auto var = NormalVariable!double(0, 1);
-    auto sample1 = threadLocalPtr!Random.randomSlice(var, 5, 3);
-
-    //Using alias of default thread local Random Engine (`rne`):
-    auto sample2 = randomSlice!cdouble(var, 5, 3);
-
-    Random rng = Random(12345);
-    auto sample3 = rng.randomSlice!cdouble(var, 5, 3); //rng safely passed by ref
-}
-
-/// Random binary data
-nothrow @safe version(mir_random_test) unittest
+pragma(inline, false)
+auto randomSlice(T, G, size_t N)(G gen, size_t[N] lengths...)
```

Comment:
It might be worth mentioning in the documentation that for this family of functions, if `T` is floating point each result will be in the interval `(-1.0, 1.0)`.